clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -438,16 +438,20 @@ private static FileChannel openChannel(File file,
                                            int initFileSize,
                                            boolean preallocate) throws IOException {
         if (mutable) {
-            if (fileAlreadyExists || !preallocate) {
-                return FileChannel.open(file.toPath(), StandardOpenOption.CREATE, StandardOpenOption.READ,
-                        StandardOpenOption.WRITE);
+            if (fileAlreadyExists) {
+                return FileChannel.open(file.toPath(), StandardOpenOption.WRITE, StandardOpenOption.READ);
             } else {
```

Comment:
Instead of having an `else` with a nested `if/else`, we could keep it flat:

```scala
if (fileAlreadyExists)
...
else if (preallocate)
...
else
...
```